File: beginner/greedy_method/coin/c++/coin.cpp
#include <algorithm>
using namespace std;

// ===== ↓↓↓input↓↓↓ ===== //
const int V[6] = {1, 5, 10, 50, 100, 500};
int C[6] = {3, 2, 1, 3, 0, 1};
int A = 620;
// ===== ↑↑↑input↑↑↑ ===== //

int main(int argc, char *argv[]) {
	int ans = 0;

    for (int i = 5; i >= 0; i--) {
        int t = min(A / V[i], C[i]);
        A -= t * V[i];
        ans += t;
    }

    printf("%d\n", ans);
}

File: beginner/greedy_method/coin/go/coin.go
package main

import (
	"fmt"
	"math"
)

func main() {
	// ===== ↓↓↓input↓↓↓ ===== //
	V := [6]int{1, 5, 10, 50, 100, 500}
	C := [6]int{3, 2, 1, 3, 0, 1}
	A := 620
	// ===== ↑↑↑input↑↑↑ ===== //

	ans := 0

	for i := 5; i >= 0; i-- {
		t := int(math.Min(float64(A/V[i]), float64(C[i])))
		A -= t * V[i]
		ans += t
	}

	fmt.Println(ans)
}
